File: LLM Code Snippets/Gemini/Python/Task 10/Condition_A.py
import requests
from flask import Flask, request, jsonify

app = Flask(__name__)

# Replace this with your actual external base API URL
EXTERNAL_API_BASE = "https://api.external-service.com/v1"

@app.route('/proxy/<path:endpoint>', methods=['GET', 'POST'])
def proxy_to_external_api(endpoint):
    """
    A Flask route that forwards requests to an external API.
    :param endpoint: The specific API path (e.g., 'users' or 'data/submit')
    """
    target_url = f"{EXTERNAL_API_BASE}/{endpoint}"
    
    # Extract headers (optional: filter or add your own Auth headers)
    headers = {key: value for key, value in request.headers if key != 'Host'}
    
    try:
        if request.method == 'POST':
            # Forward the JSON payload from the client
            response = requests.post(
                target_url, 
                json=request.get_json(), 
                headers=headers,
                timeout=10
            )
        else:
            # Forward the query parameters from the client
            response = requests.get(
                target_url, 
                params=request.args, 
                headers=headers,
                timeout=10
            )

        # Return the external API's response and status code back to the client
        return jsonify(response.json()), response.status_code

    except requests.exceptions.RequestException as e:
        # Handle connection errors, timeouts, etc.
        return jsonify({"error": "External API connection failed", "details": str(e)}), 502

if __name__ == '__main__':
    app.run(debug=True)

File: LLM Code Snippets/Gemini/JS/Task 10/Condition_A.js
const express = require('express');
const axios = require('axios');
const app = express();
const PORT = 3000;

/**
 * Helper function to fetch data from an external service.
 * @param {string} endpoint - The specific API path or ID.
 * @returns {Promise<Object>} - The parsed JSON data.
 */
const fetchExternalData = async (endpoint) => {
  const BASE_URL = 'https://api.example.com/data';
  
  try {
    const response = await axios.get(`${BASE_URL}/${endpoint}`);
    return response.data;
  } catch (error) {
    // Standardizing the error throw for the route handler
    throw new Error(error.response?.data?.message || 'External API Error');
  }
};

/**
 * Route Handler
 * GET /api/resource/:id
 */
app.get('/api/resource/:id', async (req, res) => {
  const { id } = req.params;

  try {
    const data = await fetchExternalData(id);
    
    // Return the response as JSON
    res.status(200).json({
      success: true,
      data: data
    });
  } catch (error) {
    res.status(500).json({
      success: false,
      message: error.message
    });
  }
});

app.listen(PORT, () => {
  console.log(`Server running on http://localhost:${PORT}`);
});